### DISCUSSION TITLE
通过processor_split_key_value 方式采集数据的时候，会显示content

### DISCUSSION THREAD

**Discussion by bad-youngster** [alibaba/ilogtail#555] [Help] [answered]
通过processor_split_key_value获取数据，还会有centent包裹数据
![image](https://user-images.githubusercontent.com/19181456/206334740-bc7de5e8-36ed-48b1-ba98-23b4c963ba38.png)

第二个疑问，配置是采用空格的方式做分割符，value 被“” 包裹了，也被分割了，例如http_user_agent这个参数

![image](https://user-images.githubusercontent.com/19181456/206334895-e202d359-f984-4981-86c4-5938ea88f070.png)

yaml 配置分割

processors:
  - Type: processor_split_key_value
    SourceKey: content
    Delimiter: " "
    Separator: ":"

**Comment by Takuka0311** [comment, marked as answer]
1. 关于content，可以配置KeepSource参数，参考https://github.com/alibaba/ilogtail/blob/main/docs/cn/data-pipeline/processor/processor-split-key-value.md
2. 这个问题processor_split_key_value插件不太支持解决。processor_split_char插件支持引用符，或许可以参考一下，然后增强processor_split_key_value插件的功能

**Comment by bad-youngster** [comment]
回答第二个问题，我的数据类型的是key:value模式的，那么processor_split_char不支持的吧

**Reply by Takuka0311** [reply]
是的，processor_split_char不支持kv模式。若是想用引用符功能，或许需要参考processor_split_char来修改processor_split_key_value的代码

**Reply by Takuka0311** [reply]
也可以看一看，是否多个插件的组合能够达到目标

**Reply by bad-youngster** [reply]
”参考processor_split_char来修改processor_split_key_value的“，这个有什么文档嘛

**Reply by Takuka0311** [reply]
有一个开源插件开发引导 https://github.com/alibaba/ilogtail/blob/main/docs/cn/developer-guide/plugin-development/plugin-development-guide.md
另外就是源码了，两个插件都在这里https://github.com/alibaba/ilogtail/tree/main/plugins/processor/split

**Reply by bad-youngster** [reply]
好的，有空弄下，现在还有的别的问题，大规模还没有部署下去

**Comment by pj1987111** [comment]
我可以来改下

**Comment by pj1987111** [comment]
已支持引用，可以参考 #762 
文档已更新
